lib/charms/ip_router_interface/v0/ip_router_interface.py

```diff
@@ -89,7 +93,7 @@ def __init__(self, *args):
         self.RouterRequirer = RouterRequires(charm=self)
         self.framework.observe(self.on.install, self._on_install)
         self.framework.observe(self.on.ip_router_relation_joined, self._on_relation_joined)
-        self.framework.observe(self.RouterProvider.on.routing_table_updated, self._routing_table_updated)
+        self.framework.observe(self.RouterRequirer.on.routing_table_updated, self._routing_table_updated)
 
```

Comment:
The methods below use `self.RouterRequirer.get_all_networks()` which does not exist anymore

Reply:
Fixed.